### ISSUE TITLE
Contradictory condition in encode.c

### ISSUE THREAD

**Issue by nielsdos** [lexbor/lexbor#239]
In the following code:

https://github.com/lexbor/lexbor/blob/f0539d137a9760d724dd378fad1c9fba1b083419/source/lexbor/encoding/encode.c#L386-L403

We first check if `cp < 0x80`, but there is a switch case that checks for cp to be equal to `0x203E`. This seems impossible and I guess it may be a bug. Unfortunately I'm not familiar with this encoding so I don't know what it's supposed to be instead.

**Comment by lexborisov** [comment]
@nielsdos 

In  [encoding spec](https://encoding.spec.whatwg.org/#iso-2022-jp-encoder) step 5.
```
5. If ISO-2022-JP encoder state is Roman and code point is an ASCII code point, excluding U+005C and U+007E, or is U+00A5 or U+203E, then:
    1. If code point is an ASCII code point, return a byte whose value is code point.
    2. If code point is U+00A5, return byte 0x5C.
    3. If code point is U+203E, return byte 0x7E.
```
I think I'm confused about the description of the steps.   There is indeed a logical bug here, it's strange that my static analyzers didn't detect it.
I'll do a patch on this tomorrow, with tests. 

Thanks for the report.

**Comment by lexborisov** [comment]
@nielsdos 

Thanks for the report! Fixed in [48afb62](https://github.com/lexbor/lexbor/commit/48afb625af9178d1929ec81349f173116b136310).

**Comment by nielsdos** [comment]
Thanks, I checked and the patch looks good to me.
